sections/multicolumn.liquid

```diff
@@ -80,18 +80,39 @@
                       style="padding-bottom: {{ 1 | divided_by: highest_ratio | times: 100 }}%;"
                     {% endif %}
                   >
-                    {%- capture sizes -%}
-                      (min-width: 990px) {% if section.blocks.size <= 2 %}710px{% else %}550px{% endif %}, (min-width:
-                      750px) {% if section.blocks.size == 1 %}710px{% else %}550px{% endif %}, calc(100vw - 30px)
+                    {%- liquid
+                      assign numberOfColumns = section.settings.columns_desktop
+                      assign numberOfColumnsMobile = section.settings.columns_mobile
+                      assign maxImageWidth = settings.page_width | times: 2
+                      if section.settings.image_width == 'half'
+                        assign imageWidth = 0.5
+                      elsif section.settings.image_width == 'third'
+                        assign imageWidth = 0.33
+                      else
+                        assign imageWidth = 1
+                      endif
+                    -%}
+                    {%- capture widths -%}
+                      {{ settings.page_width | times: imageWidth | divided_by: numberOfColumns | round: 0 | times: 2 }},
+                      {{ settings.page_width | times: imageWidth | divided_by: numberOfColumns | round: 0 }},
+                      {{ 989 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
+                      {{ 989 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }},
+                      {{ 749 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
+                      {{ 749 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}
+                      {{ 374 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
+                      {{ 374 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}
```

Comment:
I see the approach here (374 up to first breakpoint, 749 up to second breakpoint, etc.) but I don't think we need to be quite so targetted with the widths. The `sizes` are the important part. As long as there are enough sizes to cover most cases then we are good - the browser will select the best one based on `sizes`.

I would replace with the actual breakpoints to make it simple:
```
                      {{ 990 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
                      {{ 990 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }},
                      {{ 750 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
                      {{ 750 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}
                      {{ 375 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 | times: 2 }},
                      {{ 375 | times: imageWidth | divided_by: numberOfColumnsMobile | round: 0 }}
```

Or you could even do something arbitrary like `50, 75, 100, 150, 200, 300, 400, 500, 750, 1000, 1500, 2000` which will always have a close width.

Reply:
Missing a comma in one of these.

Reply:
I don't like hardcoded widths that aim to be accurate (what we historically do). They take effort to decide on and are easily rendered obsolete with the tweak of one setting or css style. But I could be sold on the arbitrary hard coded list Stu suggested because it can more or less be repurposed to all sections without having to give the values a second thought.

With the current approach you have here the values are at least dynamic and not made obsolete as easily. It also takes the dozens of possible combinations of variables and providing fairly accurate widths without needing to actually add dozens of hard coded values and noise to the dom.

Though there a few things I don't really like now that I'm seeing this approach in practice..
- We can end up with kind of redundant values that are only a few pixels apart
- We can't really ensure these are in a sensible order so the dom output is actually pretty difficult to parse while debugging (maybe with more liquid shenanigans)
- More cognitive load in the liquid vs a simple comma separated string